blocks/api/factory.js

```diff
@@ -116,3 +121,19 @@ export function switchToBlockType( block, name ) {
 		uid: index === firstSwitchedBlock ? block.uid : result.uid,
 	} ) );
 }
+
+/**
+ * Creates a new reusable block.
+ *
+ * @param {String} type       The type of the block referenced by the reusable block
+ * @param {Object} attributes The attributes of the block referenced by the reusable block
+ * @return {Object}           A reusable block object
+ */
+export function createReusableBlock( type, attributes ) {
+	return {
+		id: uuid(),
```

Comment:
The apparent overlap yet disconnect between a reusable block's `id` property and a standard block's `uid` property is unfortunate. I can see why we might want to call this one `id`, since it aligns well with how it is modeled on the server (the post's `id`), but it makes me wonder whether we should then align the standard block property to `id` as well. I'm not strongly attached to `uid` as a property name.

Do you have any thoughts?

Reply:
I'd be into aligning them so long as we're explicit when writing code that interacts with both types of ID, e.g:

```js
const blockId = block.id;
const reusableBlockId = block.attributes.ref;
```

This would be a breaking change though, since there might be third party code that references `block.uid`.